test/test_ops.py

```diff
@@ -696,7 +706,7 @@ def _extract_data_ptrs(out):
                 return tuple(map(lambda t: t.data_ptr(), out))
 
             def _compare_out(transform, *, compare_strides_and_data_ptrs=True):
-                out = _apply_out_transform(transform, expected)
+                out = _apply_empty_like(expected)
```

Comment:
you are not applying expected transforms here, and thus are not testing what this test is expected to test.

Reply:
Thanks for the quick review @ngimel! 

I definitely could be wrong, as this is my first PR to this repo! if so, thanks for your patience and could you please help me understand the code a bit better? I thought that the transform was applied on lines 713 and the assert was on 716

```py
                op_out(out=out)
                final_strides = _extract_strides(out)
                final_ptrs = _extract_data_ptrs(out)
                self.assertEqual(expected, out)
```

and `expected` is gotten on line 649 `expected = op(sample.input, *sample.args, **sample.kwargs)`  so calling `out = _apply_out_transform(transform, expected)` looked to me like it is calling `out = op(op(sample.input))` and that is why there are errors for ops that change the strides.

Reply:
`_compare_out` applies different transforms to `expected` values (see `_case_zero_transform`, `_case_one_transform`, `_cast_two_transform` to test different `out` behaviors, see the comments in the code. You are replacing all them with a single `empty_like` transform.

Reply:
You are right @ngimel, I think i am barking up the wrong tree.  Can you confirm that the existing 'expected failures' are expected? for example i can reproduce the failure for `test_out_nn_functional_avg_pool3d`: But I do think that my `in_place`  and `expected` tensors should be equal in the following code:

```py
In [1]: import torch

In [2]: t = torch.arange(2*3*3*4*4.0).reshape(2, 3, 3, 4, 4)

In [3]: expected = torch.nn.functional.avg_pool3d(t, (2,2,2))

In [4]: in_place = torch.zeros([i*2 for i in expected.shape], dtype=expected.dtype)[::2,::2,::2,::2,::2]

In [5]: torch.nn.functional.avg_pool3d(t, (2,2,2), out=in_place)
Out[5]: 
tensor([[[[[ 10.5000,  18.5000],
           [106.5000, 114.5000]]],

         [[[  0.0000,   0.0000],
           [  0.0000,   0.0000]]],

         [[[  0.0000,   0.0000],
           [  0.0000,   0.0000]]]],

        [[[[  0.0000,   0.0000],
           [  0.0000,   0.0000]]],

         [[[  0.0000,   0.0000],
           [  0.0000,   0.0000]]],

         [[[  0.0000,   0.0000],
           [  0.0000,   0.0000]]]]])

In [6]: expected
Out[6]: 
tensor([[[[[ 10.5000,  12.5000],
           [ 18.5000,  20.5000]]],

         [[[ 58.5000,  60.5000],
           [ 66.5000,  68.5000]]],

         [[[106.5000, 108.5000],
           [114.5000, 116.5000]]]],

        [[[[154.5000, 156.5000],
           [162.5000, 164.5000]]],

         [[[202.5000, 204.5000],
           [210.5000, 212.5000]]],

         [[[250.5000, 252.5000],
           [258.5000, 260.5000]]]]])

```

Reply:
Yep, I think the operator issues are real (and as best I can tell, the test is correct)